### ISSUE TITLE
emptyList parameter becomes null

### ISSUE THREAD

**Issue by coullin** [OpenFeign/feign#1224]
After migrating from Hoxton.SR1 to Hoxton.SR4, my code started breaking due to change in the way empty list parameter is encoded by OpenFeign.

Here is my Feign Interface

```
@RequestMapping(value = ["/api/order"], method = [RequestMethod.GET])
fun findAllOrders(@RequestParam(value = "status", required = false) statusList: List<String>): RestResponsePage<OrderDto>
```
When invoking this client, if i pass emptyList() as the input, Feign sets parameter value for status param as null

`https://<host:port>/api/order?status=null`

This null value starts breaking the target server with 400 BAD request. Till Hoxton.SR1, the request in Feign Logs was showing up as:

`https://<host:port>/api/order?status`

Which worked perfectly well.

Is there some change in configuration required to restore the previous behavior after migrating to Spring Cloud Hoxton.SR4

**Comment by kdavisk6** [comment]
@coullin 

First, I recommend that you check with the Spring Cloud team to determine which version of OpenFeign is being included in Spring Cloud OpenFeign under Hoxton.SR4 and then let us know here.

Second, take a look at how we handle [Undefined vs. Empty Values](https://github.com/OpenFeign/feign#undefined-vs-empty-values) in our documentation.  This may help.

**Comment by coullin** [comment]
well then i fixed by this work around below, return null instead return 'null' string when expandIterable returns null .

`@Override
    String expand(Object variable, boolean encode) {
      StringBuilder expanded = new StringBuilder();
      if (Iterable.class.isAssignableFrom(variable.getClass())) {
        String exp = this.expandIterable((Iterable<?>) variable);
        if(null == exp) {
          return null;
        }
        expanded.append(exp);
      } else {
        expanded.append((encode) ? encode(variable) : variable);
      }

      /* return the string value of the variable */
      String result = expanded.toString();
      if (!this.matches(result)) {
        throw new IllegalArgumentException("Value " + expanded
            + " does not match the expression pattern: " + this.getPattern());
      }
      return result;
    }
`

**Comment by kdavisk6** [comment]
This was been corrected #1200
